### PR TITLE
Command mapping 5502 1

### PR DESCRIPTION
**Pull request by jayvikramsharma1**
### :nut_and_bolt: Description: What code changed, and why?
command mapping for following commands
**gather-logs --> chef-automate gather-logs **

Usage:
    chef-automate gather-logs

Options:
    -o, --ssh-options OPTIONS          Additional options to pass to ssh
    -c, --sudo-command SUDO_COMMAND    Alternate command for sudo (default: "sudo")
    -s, --sudo-options SUDO_OPTIONS    Additional options for remote sudo
    -r, --remote-log-dir DIRECTORY     Where to temporarily store remote log files (default: "/var/tmp")
    -l, --local-log-dir DIRECTORY      Where to store the logs locally (default: "/var/tmp")
    -n, --log-lines NUMBER             How many journalctl lines to capture
    -h, --help                         print help
    -v, --verbose                      Enable verbose output
    -c, --config-file CONFIG_FILE      Load settings from config file (default: "/root/workspace/a2-ha-backend/a2ha.rb")

** status --> chef-automate status **

Usage:
    chef-automate status
	shows status of services running.

** test --> chef-automate test **
used to run automate ha smoke test cases manually

### :chains: Related Resources

https://app.zenhub.com/workspaces/the-delta-quadrant-5fc7c4c3922dc10021323fa6/issues/chef/automate/5502

### :+1: Definition of Done

Dev Done, coding and dev manual testing

### :athletic_shoe: How to Build and Test the Change

rebuild component/automate-cli

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/status-ha.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"bytes"
+	"os"
+	"os/exec"
+
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+)
+
+func executeHAStatus(args []string) error {
+	writer.Printf("getting status \n")
```
Comment (punitmundra) [inline components/automate-cli/cmd/chef-automate/status-ha.go:12]: getting status -> A2HA Cluster Status

**Review by punitmundra — APPROVED** [review]
just change the one of the logs -> A2HA Cluster status. rest every thing is ok. 
Add in the desc...that a2ha gather-logs only work ...when we have deployed cluster.

File: components/automate-cli/cmd/chef-automate/gather-logs-ha-template.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Usage:
+    chef-automate gather-logs
+
+Options:
+    -o, --ssh-options OPTIONS          Additional options to pass to ssh
+    -c, --sudo-command SUDO_COMMAND    Alternate command for sudo (default: "sudo")
+    -s, --sudo-options SUDO_OPTIONS    Additional options for remote sudo
+    -r, --remote-log-dir DIRECTORY     Where to temporarily store remote log files (default: "/var/tmp")
+    -l, --local-log-dir DIRECTORY      Where to store the logs locally (default: "/var/tmp")
+    -n, --log-lines NUMBER             How many journalctl lines to capture
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/gather-logs-ha-template.go:13]: journalctl is a command. Are we trying to capture the command output or the contents of journal logs?

File: components/automate-cli/cmd/chef-automate/gather-logs-ha-template.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    -h, --help                         print help
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/gather-logs-ha-template.go:14]:
```suggestion
    -h, --help                         Print help
```

File: components/automate-cli/cmd/chef-automate/gather-logs-ha-template.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    -v, --verbose                      Enable verbose output
+    -c, --config-file CONFIG_FILE      Load settings from config file (default: "/root/workspace/a2-ha-backend/a2ha.rb")
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/gather-logs-ha-template.go:16]: Is it `/root` or `/hab` @punitmundra ?

File: components/automate-cli/cmd/chef-automate/statusHaHelpDoc.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package main
+
+var statusHAHelpDocs = `
+Usage:
+    chef-automate status
+	shows status of services running.
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/statusHaHelpDoc.go:6]: If it is across all the nodes better to mention that.

File: components/automate-cli/cmd/chef-automate/status.go [hunk 6]
```diff
@@ -54,87 +54,92 @@ var (
...
+			}
 		}
-	}
 
-	getStatus := func() (*api.StatusResponse, error) {
-		connection, err := client.Connection(client.DefaultClientTimeout)
-		if err != nil {
-			return nil, err
-		}
+		getStatus := func() (*api.StatusResponse, error) {
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/status.go:72]: Are these tabs? If so please convert to whitespaces

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/gather-logs.go [hunk 7]
```diff
@@ -115,28 +114,33 @@ var gatherLogsCmdFlags = struct {
...
-	// Ensure we can write to any user given log locations
-	overridePath := ""
-	if len(args) > 0 && args[0] != "" {
-		var err error
-		overridePath, err = expandOutFilePath(args[0])
-		if err != nil {
-			return err
+	if isA2HADeployment() {
+		return executeHAGatherLogsA2HA(args)
+	} else {
```
Comment (vivek-yadav) [inline components/automate-cli/cmd/chef-automate/gather-logs.go:119]: removing Else will make the flow simple, also diff will be smaller.

File: components/automate-cli/cmd/chef-automate/status.go [hunk 8]
```diff
@@ -54,87 +54,92 @@ var (
...
-			"Status from deployment with channel [%s] and type [%s]",
-			res.DeploymentConfig.V1.Svc.Channel.GetValue(),
-			res.DeploymentConfig.V1.Svc.DeploymentType.GetValue(),
-		)
-		writer.Title(res.ServiceStatus.FormatStatus())
-		status.GlobalResult = statusResult{
-			Services: res.ServiceStatus.FormattedServices(),
+	if isA2HADeployment() {
+		return executeHAStatus(args)
+	} else {
```
Comment (vivek-yadav) [inline components/automate-cli/cmd/chef-automate/status.go:59]: Same here, if Else is remove the flow will be simple, and diff will be smaller

File: components/automate-cli/cmd/chef-automate/testHA.go [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+		c.Stdout = &out
+		c.Stderr = &stderr
+		err := c.Run()
+		if err != nil {
+			writer.Printf(stderr.String())
+			return status.Wrap(err, status.CommandExecutionError, "")
+		}
+		writer.Print(out.String())
+		return err
+	} else {
```
Comment (vivek-yadav) [inline components/automate-cli/cmd/chef-automate/testHA.go:46]: same here, Else is not needed.

**Pull request closed without merging**